CMakeLists.txt

```diff
@@ -333,7 +333,37 @@ if(GCC OR CLANG)
   # TODO(CryptoAlg-759): enable '-Wpedantic' if awslc has to follow c99 spec.
   if(CLANG OR (GCC AND CMAKE_C_COMPILER_VERSION VERSION_GREATER "4.1.3"))
     # GCC 4.1.3 and below do not support all of these flags or they raise false positives.
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden -Wall -Wextra -Wno-unused-parameter -Werror")
+    if(MSVC)
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-bad-function-cast")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-cast-function-type-strict")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-cast-function-type")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-cast-align")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-cast-qual")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-comma")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-c++98-compat")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-declaration-after-statement")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-disabled-macro-expansion")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-deprecated-declarations")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-extra-semi-stmt")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-extra-semi")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-implicit-int-conversion")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-missing-noreturn")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-missing-variable-declarations")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-overlength-strings")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-nonportable-system-include-path")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-reserved-identifier")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-shorten-64-to-32")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-sign-conversion")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unreachable-code-break")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-tautological-value-range-compare")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unreachable-code-return")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-unsafe-buffer-usage")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-unused-macros")
+      set(C_CXX_FLAGS "${C_CXX_FLAGS} -Wno-used-but-marked-unused")
+    else()
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
```

Comment:
Is the intent to remove this for Windows/MSVC? Won't this expose all hidden symbols when built with MSVC?

Reply:
This flag is not supported by `clang-cl` so it must be excluded.

Reply:
Is there a way to differentiate `clang-cl` with MSVC? I'm not sure if we want to disable this for all of Windows entirely, do we understand the repercussions of turning this flag off for MSVC?

Reply:
This block is unique to `clang-cl` since it satisfies both the `CLANG` and `MSVC` -- this block doesn't affect "normal (`cl`) MSVC builds.

Reply:
Ahhhh thanks for clearing things up there. I didn't know `clang-cl` was Clang/MSVC hybrid